### ISSUE TITLE
Constant y_pred values

### ISSUE THREAD

**Issue by gaolong** [bernardo-de-almeida/DeepSTARR#3]
Hi,

I was trying to run DeepSTARR_training.ipynb, but spearman correlations are always nan suggesting a constant input values. Also it stops at 14 epoch, please see the information below. Have any idea why? Or I just ignore it and run downstream analysis

Tensor("Dense_Dev_5/BiasAdd:0", shape=(None, 1), dtype=float32)
Tensor("Dense_Dev_target_5:0", shape=(None, None), dtype=float32)
Tensor("Dense_Hk_5/BiasAdd:0", shape=(None, 1), dtype=float32)
Tensor("Dense_Hk_target_5:0", shape=(None, None), dtype=float32)
Train on 402296 samples, validate on 40570 samples
Epoch 1/100
402296/402296 [==============================] - 67s 168us/step - loss: 5.0083 - Dense_Dev_loss: 2.3143 - Dense_Hk_loss: 2.6940 - Dense_Dev_Spearman: nan - Dense_Hk_Spearman: nan - val_loss: 5.1719 - val_Dense_Dev_loss: 2.3108 - val_Dense_Hk_loss: 2.8607 - val_Dense_Dev_Spearman: nan - val_Dense_Hk_Spearman: nan
Epoch 2/100
402296/402296 [==============================] - 66s 164us/step - loss: 5.0052 - Dense_Dev_loss: 2.3113 - Dense_Hk_loss: 2.6939 - Dense_Dev_Spearman: nan - Dense_Hk_Spearman: nan - val_loss: 5.1691 - val_Dense_Dev_loss: 2.3082 - val_Dense_Hk_loss: 2.8605 - val_Dense_Dev_Spearman: nan - val_Dense_Hk_Spearman: nan
Epoch 3/100
402296/402296 [==============================] - 68s 168us/step - loss: 5.0052 - Dense_Dev_loss: 2.3113 - Dense_Hk_loss: 2.6939 - Dense_Dev_Spearman: nan - Dense_Hk_Spearman: nan - val_loss: 5.1697 - val_Dense_Dev_loss: 2.3089 - val_Dense_Hk_loss: 2.8604 - val_Dense_Dev_Spearman: nan - val_Dense_Hk_Spearman: nan
Epoch 4/100
402296/402296 [==============================] - 68s 169us/step - loss: 5.0052 - Dense_Dev_loss: 2.3112 - Dense_Hk_loss: 2.6940 - Dense_Dev_Spearman: nan - Dense_Hk_Spearman: nan - val_loss: 5.1678 - val_Dense_Dev_loss: 2.3070 - val_Dense_Hk_loss: 2.8605 - val_Dense_Dev_Spearman: nan - val_Dense_Hk_Spearman: nan
Epoch 5/100
402296/402296 [==============================] - 68s 168us/step - loss: 5.0052 - Dense_Dev_loss: 2.3113 - Dense_Hk_loss: 2.6939 - Dense_Dev_Spearman: nan - Dense_Hk_Spearman: nan - val_loss: 5.1708 - val_Dense_Dev_loss: 2.3099 - val_Dense_Hk_loss: 2.8605 - val_Dense_Dev_Spearman: nan - val_Dense_Hk_Spearman: nan
Epoch 6/100
402296/402296 [==============================] - 67s 168us/step - loss: 5.0052 - Dense_Dev_loss: 2.3113 - Dense_Hk_loss: 2.6939 - Dense_Dev_Spearman: nan - Dense_Hk_Spearman: nan - val_loss: 5.1683 - val_Dense_Dev_loss: 2.3068 - val_Dense_Hk_loss: 2.8611 - val_Dense_Dev_Spearman: nan - val_Dense_Hk_Spearman: nan
Epoch 7/100
402296/402296 [==============================] - 68s 168us/step - loss: 5.0052 - Dense_Dev_loss: 2.3113 - Dense_Hk_loss: 2.6939 - Dense_Dev_Spearman: nan - Dense_Hk_Spearman: nan - val_loss: 5.1713 - val_Dense_Dev_loss: 2.3101 - val_Dense_Hk_loss: 2.8608 - val_Dense_Dev_Spearman: nan - val_Dense_Hk_Spearman: nan
Epoch 8/100
402296/402296 [==============================] - 70s 175us/step - loss: 5.0052 - Dense_Dev_loss: 2.3113 - Dense_Hk_loss: 2.6939 - Dense_Dev_Spearman: nan - Dense_Hk_Spearman: nan - val_loss: 5.1696 - val_Dense_Dev_loss: 2.3086 - val_Dense_Hk_loss: 2.8607 - val_Dense_Dev_Spearman: nan - val_Dense_Hk_Spearman: nan
Epoch 9/100
402296/402296 [==============================] - 69s 172us/step - loss: 5.0052 - Dense_Dev_loss: 2.3113 - Dense_Hk_loss: 2.6939 - Dense_Dev_Spearman: nan - Dense_Hk_Spearman: nan - val_loss: 5.1700 - val_Dense_Dev_loss: 2.3079 - val_Dense_Hk_loss: 2.8617 - val_Dense_Dev_Spearman: nan - val_Dense_Hk_Spearman: nan
Epoch 10/100
402296/402296 [==============================] - 70s 173us/step - loss: 5.0052 - Dense_Dev_loss: 2.3112 - Dense_Hk_loss: 2.6939 - Dense_Dev_Spearman: nan - Dense_Hk_Spearman: nan - val_loss: 5.1680 - val_Dense_Dev_loss: 2.3073 - val_Dense_Hk_loss: 2.8604 - val_Dense_Dev_Spearman: nan - val_Dense_Hk_Spearman: nan
Epoch 11/100
402296/402296 [==============================] - 70s 174us/step - loss: 5.0052 - Dense_Dev_loss: 2.3113 - Dense_Hk_loss: 2.6939 - Dense_Dev_Spearman: nan - Dense_Hk_Spearman: nan - val_loss: 5.1712 - val_Dense_Dev_loss: 2.3104 - val_Dense_Hk_loss: 2.8604 - val_Dense_Dev_Spearman: nan - val_Dense_Hk_Spearman: nan
Epoch 12/100
402296/402296 [==============================] - 70s 173us/step - loss: 5.0052 - Dense_Dev_loss: 2.3113 - Dense_Hk_loss: 2.6939 - Dense_Dev_Spearman: nan - Dense_Hk_Spearman: nan - val_loss: 5.1686 - val_Dense_Dev_loss: 2.3078 - val_Dense_Hk_loss: 2.8604 - val_Dense_Dev_Spearman: nan - val_Dense_Hk_Spearman: nan
Epoch 13/100
402296/402296 [==============================] - 69s 173us/step - loss: 5.0052 - Dense_Dev_loss: 2.3113 - Dense_Hk_loss: 2.6939 - Dense_Dev_Spearman: nan - Dense_Hk_Spearman: nan - val_loss: 5.1706 - val_Dense_Dev_loss: 2.3099 - val_Dense_Hk_loss: 2.8604 - val_Dense_Dev_Spearman: nan - val_Dense_Hk_Spearman: nan
Epoch 14/100
402296/402296 [==============================] - 69s 173us/step - loss: 5.0052 - Dense_Dev_loss: 2.3113 - Dense_Hk_loss: 2.6939 - Dense_Dev_Spearman: nan - Dense_Hk_Spearman: nan - val_loss: 5.1704 - val_Dense_Dev_loss: 2.3092 - val_Dense_Hk_loss: 2.8608 - val_Dense_Dev_Spearman: nan - val_Dense_Hk_Spearman: nan

Thanks,
Long

**Comment by gaolong** [comment]
It looks like it always output constant values.
<img width="420" alt="deepstarr_test" src="https://user-images.githubusercontent.com/7356655/180142416-e69a93f3-a8a6-4bc3-8944-347cab2af3df.PNG">

**Comment by bernardo-de-almeida** [comment]
Hi, 

Usually this happens when the model is not able to learn, possibly because the data are unpredictable. Are you using the DeepSTARR data as input or your own data?

It stops at 14 epochs because the validation loss didn't improve by 10 epochs. See early_stop argument.

Best,
Bernardo

**Comment by gaolong** [comment]
Hi Bernardo,

Thanks for your response. Yes, I did use the same data you guys included in the notebook shown below. Also I didn't modify any code in the notebook, so I was a little surprised I couldn't successfully train the model. The only difference is the version of tf and keras. I installed a relatively new version of tf/keras. Maybe I missed something during training.

Thanks,
Long

[../](https://data.starklab.org/almeida/DeepSTARR/)
[Sequences_All.fa](https://data.starklab.org/almeida/DeepSTARR/Data/Sequences_All.fa)                                   22-Jan-2021 17:41           140709806
[Sequences_Test.fa](https://data.starklab.org/almeida/DeepSTARR/Data/Sequences_Test.fa)                                  22-Jan-2021 17:40            12025132
[Sequences_Train.fa](https://data.starklab.org/almeida/DeepSTARR/Data/Sequences_Train.fa)                                 22-Jan-2021 17:40           116924268
[Sequences_Val.fa](https://data.starklab.org/almeida/DeepSTARR/Data/Sequences_Val.fa)                                   22-Jan-2021 17:40            11760406
[Sequences_activity_All.txt](https://data.starklab.org/almeida/DeepSTARR/Data/Sequences_activity_All.txt)                         22-Jan-2021 17:41            52567661
[Sequences_activity_Test.txt](https://data.starklab.org/almeida/DeepSTARR/Data/Sequences_activity_Test.txt)                        22-Jan-2021 17:40             4476205
[Sequences_activity_Train.txt](https://data.starklab.org/almeida/DeepSTARR/Data/Sequences_activity_Train.txt)                       22-Jan-2021 17:40            43682499
[Sequences_activity_Val.txt](https://data.starklab.org/almeida/DeepSTARR/Data/Sequences_activity_Val.txt)                         22-Jan-2021 17:40             4409299

**Comment by bernardo-de-almeida** [comment]
Can you send me your environment information for all modules such that I could try to reproduce your problem?
Thanks!

**Comment by gaolong** [comment]
It looks like tensorflow version was the problem. After I switched tensorflow 2 to tensorflow 1.14, I could reproduce your results. Thanks again!

**Comment by bernardo-de-almeida** [comment]
Great, thanks for the feedback!
